task/aontimer/src/main.rs

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+use aontimer as aon;
+use drv_riscv_plic_api::*;
+use userlib::*;
+
+task_slot!(INT_CONTROLLER, ext_int_ctrl);
+
+const AONTIMER_BARK: u32 = 0x1;
+const AONTIMER_TIME: u32 = 0x2;
+
+const BARK_INT: u32 = 0;
+
+fn sleep_and_listen(timer: &aon::AonTimer) {
+    sys_log!("Sleeping...");
+    let alarm = sys_get_timer().now;
+    sys_set_timer(Some(alarm + 1000), AONTIMER_TIME);
+    loop {
+        let result = sys_recv_closed(
+            &mut [],
+            AONTIMER_BARK | AONTIMER_TIME,
+            TaskId::KERNEL,
+        )
+        .unwrap();
+        if result.operation & AONTIMER_BARK != 0x0 {
+            sys_log!("{}", sys_get_timer().now);
+            //reset the interrupt
+            if timer.bark_cb.is_some() {
+                sys_log!("calling back");
+                (timer.bark_cb.unwrap())();
+            }
+        }
+        if result.operation & AONTIMER_TIME != 0x0 {
+            // Comment the following line of code to recieve the bark, and then be forced reset.
+            return; //You've woken up
+        }
+    }
+}
+
+#[export_name = "main"]
+fn main() -> ! {
+    let int_ctrl = RiscvIntCtrl::from(INT_CONTROLLER.get_task_id());
+
+    int_ctrl.disable_int(BARK_INT).unwrap();
+
+    const AONTIMER_BASE: u32 = 0x4047_0000;
+    const AONTIMER_FREQ: u32 = 200_000;
```

Comment:
Is this frequency configurable by each task? Or is this something that is fixed for the open titan chip?

Reply:
@rivos-tkng Would know for sure, but I think this constant is used to configure the AON timer.  So it would be configurable by whatever task owns it.

Reply:
As a slightly related side question,  @FawazTirmizi, can two task have access to the same peripheral in the chip.toml?

Reply:
Yeah, it should be. I haven't tested it yet, but some of Oxide's apps like the [gimlet-rot](https://github.com/rivosinc/hubris/blob/2563c2c4550c0f0a558364aea200ddacc227ca2a/app/gimlet-rot/app.toml) do this.

Reply:
Hmm, that could cause problems with two task both trying to manage one device.  It may be an interesting feature if you could define a memory region to be exclusive to one task.  Then if two task both `use` it, it could provide a compile time error.

Reply:
I think having multiple tasks own a single peripheral isn't going to work without a lot of runtime coordination between the two. I think the way this will have to be organized is that there is a single peripheral driver and any task that wants to use the peripheral talks to the driver to get its work done.

Reply:
The timer frequency is not configurable from SW. This was supposed to be a C-style macro for converting time -> register count.

Reply:
If the frequency is fixed, it sounds like the type of thing that should be in the chip.toml

Reply:
Since it is fixed by the hardware, it seems like it should be in the chip.toml.  However, I don't know how hard it is to add arbitrary fields.  @FawazTirmizi

Reply:
I went ahead and move the timer frequency to the app.toml along with the bark/bite intervals.